block.go

```diff
@@ -260,9 +260,19 @@ func (p *Markdown) prefixHeading(data []byte) int {
 		if id == "" && p.extensions&AutoHeadingIDs != 0 {
 			id = SanitizedAnchorName(string(data[i:end]))
 		}
+
 		block := p.addBlock(Heading, data[i:end])
 		block.HeadingID = id
 		block.Level = level
+
+		if p.extensions&AutoHeadingAnchorLinks != 0 {
+			link := NewNode(Link)
+			link.Destination = []byte("#" + id)
+			linkText := NewNode(Text)
+			linkText.Literal = []byte("#")
+			link.AppendChild(linkText)
+			block.AppendChild(link)
+		}
```

Comment:
There's another section which handles headings of the `====` variety; example:

```
My Heading
==========
```

This code will need to be copied there, too. I recommend adding a function like this:

```
func anchorHeadingLink(node Node) {
	link := NewNode(Link)
	link.Destination = []byte("#" + id)
	linkText := NewNode(Text)
	linkText.Literal = []byte("#")
	link.AppendChild(linkText)
	node.AppendChild(link)
}
```